network/common/src/main/java/org/apache/spark/network/server/TransportRequestHandler.java

```diff
@@ -102,7 +93,8 @@ public void handle(RequestMessage request) {
 
   private void processFetchRequest(final ChunkFetchRequest req) {
     final String client = NettyUtils.getRemoteAddress(channel);
-    streamIds.add(req.streamChunkId.streamId);
+
+    streamManager.registerChannel(channel, req.streamChunkId.streamId);
```

Comment:
As we don't know from this point in the code whether the registerChannel method will throw an exception, let's move it inside the try-catch for the getChunk(). Otherwise we could leave a message unresponded-to, which would be bad.